files/lib/data/conversation/ConversationAction.class.php

```diff
@@ -216,6 +216,11 @@ public function markAsRead() {
 		if (empty($this->parameters['visitTime'])) {
 			$this->parameters['visitTime'] = TIME_NOW;
 		}
+
+		// userID cannot be set during an AJAX request, but only when calling this method within PHP
```

Comment:
Maybe "cannot" is the wrong wording. It should be "should not", right? So how should the code look like to get merged? Not sure right now.